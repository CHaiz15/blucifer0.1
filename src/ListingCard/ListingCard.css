.listing-card {
  margin: 3%;
  padding-bottom: 3%;
  display: flex;
  border-bottom: solid 1px black;
}

.listing-img {
  width: 35%;
  height: 200px;
  object-fit: cover;
}

.listing-info {
  margin-left: 5%;
}

.listing-cost {
  display: flex;
  flex-direction: row;
}

.price {
  font-weight: bold;
}

.details-btn {
  background-color: #002244;
  color: #fff;
  font-size: 1em;
  height: 15%;
  width: 60%;
}