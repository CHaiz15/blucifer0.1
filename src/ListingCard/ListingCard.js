import React from 'react';
import { Link } from 'react-router-dom';
import './ListingCard.css'

const ListingCard = ({ addSelectedListing, areaId, listing_id, name, cost_per_night }) => {
    const selectListing = () => {
        addSelectedListing(listing_id)
    }

    return (
        <div className='listing-card'>
            <img className='listing-img' alt='listing-img' src={process.env.PUBLIC_URL + `/images/${listing_id}_a.jpg`} />
            <section className='listing-info'>
                <h2>{name}</h2>
                <p className='listing-cost'>${cost_per_night} / Night</p>
                <Link to={`/nav/areas/${areaId}/listings/${listing_id}`}><button  className='details-btn' onClick={selectListing}>Details</button></Link>
            </section>
        </div>
    )
}

export default ListingCard;