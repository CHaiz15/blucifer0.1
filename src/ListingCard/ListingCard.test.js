import React from 'react';
import { render } from '@testing-library/react';
import ListingCard from './ListingCard';
import { shallow } from 'enzyme'

describe('ListingCard', () => {
  it('should be an instance of ListingCard component', () => {
    const wrapper = shallow(<ListingCard/>);
      expect(wrapper).toMatchSnapshot();
  });
})