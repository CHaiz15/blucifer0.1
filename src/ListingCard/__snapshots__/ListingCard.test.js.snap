// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCard should be an instance of ListingCard component 1`] = `
<div
  className="listing-card"
>
  <img
    alt="listing-img"
    className="listing-img"
    src="/images/undefined_a.jpg"
  />
  <section
    className="listing-info"
  >
    <h2 />
    <div>
      <p
        className="listing-cost-per-night"
      >
        $
         / Night
      </p>
      <Link
        to="/nav/areas/undefined/listings/undefined"
      >
        <button
          className="details-btn"
          onClick={[Function]}
        >
          Details
        </button>
      </Link>
    </div>
  </section>
</div>
`;
